Google.Authenticator/TwoFactorAuthenticator.cs

```diff
@@ -17,209 +16,222 @@ namespace Google.Authenticator
 
-        internal string GenerateHashedCode(byte[] key, long iterationNumber, int digits = 6)
+        private string GenerateHashedCode(byte[] key, long iterationNumber, int digits = 6)
         {
-            byte[] counter = BitConverter.GetBytes(iterationNumber);
+            var counter = BitConverter.GetBytes(iterationNumber);
 
             if (BitConverter.IsLittleEndian)
-            {
                 Array.Reverse(counter);
-            }
 
-            HMACSHA1 hmac = new HMACSHA1(key);
+            var hmac = new HMACSHA1(key);
 
-            byte[] hash = hmac.ComputeHash(counter);
+            var hash = hmac.ComputeHash(counter);
 
-            int offset = hash[hash.Length - 1] & 0xf;
+            var offset = hash[hash.Length - 1] & 0xf;
 
             // Convert the 4 bytes into an integer, ignoring the sign.
-            int binary =
+            var binary =
                 ((hash[offset] & 0x7f) << 24)
                 | (hash[offset + 1] << 16)
                 | (hash[offset + 2] << 8)
-                | (hash[offset + 3]);
+                | hash[offset + 3];
 
-            int password = binary % (int)Math.Pow(10, digits);
+            var password = binary % (int) Math.Pow(10, digits);
             return password.ToString(new string('0', digits));
         }
 
-        private long GetCurrentCounter()
-        {
-            return GetCurrentCounter(DateTime.UtcNow, _epoch, 30);
-        }
+        private long GetCurrentCounter() => GetCurrentCounter(DateTime.UtcNow, _epoch, 30);
 
-        private long GetCurrentCounter(DateTime now, DateTime epoch, int timeStep)
-        {
-            return (long)(now - epoch).TotalSeconds / timeStep;
-        }
+        private long GetCurrentCounter(DateTime now, DateTime epoch, int timeStep) =>
+            (long) (now - epoch).TotalSeconds / timeStep;
 
-        public bool ValidateTwoFactorPIN(string accountSecretKey, string twoFactorCodeFromClient)
-        {
-            return ValidateTwoFactorPIN(accountSecretKey, twoFactorCodeFromClient, DefaultClockDriftTolerance);
-        }
+        public bool ValidateTwoFactorPIN(
+            string accountSecretKey, 
+            string twoFactorCodeFromClient,
+            bool secretIsBase32 = false) =>
+            ValidateTwoFactorPIN(accountSecretKey, twoFactorCodeFromClient, DefaultClockDriftTolerance, secretIsBase32);
 
-        public bool ValidateTwoFactorPIN(string accountSecretKey, string twoFactorCodeFromClient, TimeSpan timeTolerance)
-        {
-            var codes = GetCurrentPINs(accountSecretKey, timeTolerance);
-            return codes.Any(c => c == twoFactorCodeFromClient);
-        }
+        public bool ValidateTwoFactorPIN(
+            string accountSecretKey,
+            string twoFactorCodeFromClient,
+            TimeSpan timeTolerance,
+            bool secretIsBase32 = false) =>
+            GetCurrentPINs(accountSecretKey, timeTolerance, secretIsBase32).Any(c => c == twoFactorCodeFromClient);

```

Comment:
I agree this is more compact, but I think it hurts legibility. I am not too bothered, but I prefer the previous form here.